### PR TITLE
cgroup metrics extension for ecs_exporter

### PR DESCRIPTION
**Pull request by JT-Alchemy**

### CODE REVIEW

File: cgroup/cgroup_collector.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+type cgroupMemoryCollector struct {
+	metrics     []cgroupMetric
+	statMetrics []statMetric
+}
+
+func NewCGroupMemoryCollector() prometheus.Collector {
+	// Initialize all desired metrics with their descriptions
+	metrics := []cgroupMetric{
```
Comment (chub) [inline cgroup/cgroup_collector.go:28]: Also:
- `memory.kmem.tcp.usage_in_bytes`
- `memory.kmem.tcp.limit_in_bytes`
- `memory.kmem.tcp.max_usage_in_bytes`
- `memory.kmem.tcp.failcnt`

File: cgroup/cgroup_collector.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+			Path:   "/sys/fs/cgroup/memory/memory.kmem.max_usage_in_bytes",
+			Metric: prometheus.NewDesc("cgroup_kmem_memory_max_usage_bytes", "Maximum memory usage of the cgroup in bytes.", nil, nil),
+		},
+		{
+			Path:   "/sys/fs/cgroup/memory/memory.kmem.limit_in_bytes",
+			Metric: prometheus.NewDesc("cgroup_kmem_memory_limit_bytes", "Memory limit of the cgroup in bytes.", nil, nil),
+		},
+		{
+			Path:   "/sys/fs/cgroup/memory/memory.kmem.failcnt",
+			Metric: prometheus.NewDesc("cgroup_kmem_memory_failcnt", "Number of memory usage hits limits.", nil, nil),
```
Comment (chub) [inline cgroup/cgroup_collector.go]: For `memory.kmem`, let's update the documentation line to say "kernel memory" (prom and grafana display these hints, so we should make them as accurate as possible"

For `memory.kmem.tcp`, it's the same for "TCP buffer memory (allocations|usage|hits)"

**Review by chub — COMMENTED** [review]
We should annotate the version somewhere too.

**Comment by JT-Alchemy** [comment]
> We should annotate the version somewhere too.

It's not in the diffs but there's a VERSION file at the top of the repo that tracks that

**Pull request merged**